packages/NodeNameResolver/NodeNameResolver.php

```diff
@@ -188,21 +197,23 @@ public function isStringName(string $resolvedName, string $desiredName): bool
             return false;
         }
 
-        // is probably regex pattern
-        if ($this->regexPatternDetector->isRegexPattern($desiredName)) {
-            return StringUtils::isMatch($resolvedName, $desiredName);
-        }
-
-        // is probably fnmatch
-        if (\str_contains($desiredName, '*')) {
-            return fnmatch($desiredName, $resolvedName, FNM_NOESCAPE);
-        }
-
         // special case
         if ($desiredName === 'Object') {
             return $desiredName === $resolvedName;
         }
 
+        if (StringUtils::isMatch($desiredName, self::CONTAINS_WILDCARD_CHARS_REGEX)) {
```

Comment:
regex can start and end as combination `(` with `)`, `<` and `>` as delimiter in php, so probably just leave the original functionality to avoid regression.

Reply:
also possible start with `{` and end with `}` as delimiter.

Reply:
these delmiter start and end are valid:

```
               '(' => ')',
                '{' => '}',
                '[' => ']',
                '<' => '>',
```

Reply:
I got the list of delimiters from `\Rector\NodeNameResolver\Regex\RegexPatternDetector::POSSIBLE_DELIMITERS`
```php
private const POSSIBLE_DELIMITERS = ['#', '~', '/'];
```

So other delimiters are not supported there already

Reply:
I see, looks good then 👍